llvm/docs/AMDGPUUsage.rst

```diff
@@ -4135,6 +4283,33 @@ Code object V5 metadata is the same as
 
      ====================== ============== ========= ================================
 
+.. _amdgpu-amdhsa-code-object-metadata-v6:
+
+Code Object V6 Metadata
++++++++++++++++++++++++
+
+.. warning::
+  Code object V6 is not the default code object version emitted by this version
+  of LLVM.
+
+
+Code object V6 metadata is the same as
+:ref:`amdgpu-amdhsa-code-object-metadata-v5` with the changes defined in table
+:ref:`amdgpu-amdhsa-code-object-metadata-map-table-v6`.
+
+  .. table:: AMDHSA Code Object V6 Metadata Map Changes
+     :name: amdgpu-amdhsa-code-object-metadata-map-table-v6
+
+     ================= ============== ========= =======================================
+     String Key        Value Type     Required? Description
+     ================= ============== ========= =======================================
+     "amdhsa.version"  sequence of    Required  - The first integer is the major
```

Comment:
Since there are no changes to the metadata do we need this? Can make the V5 one be V5 and V6.

Reply:
I anticipate that we'll want to add some more V6-only metadata at some point, that's why I just started a new  table so it's easier to follow up. I don't mind merging it with the V5 table if you really prefer

Reply:
I'd probably drop metadata version bump unless we change it.